PurchasesCoreSwift/Logging/Strings/PurchaseStrings.swift

```diff
@@ -9,15 +9,15 @@
 import Foundation
 
 // swiftlint:disable identifier_name
-@objc(RCPurchaseStrings) public class PurchaseStrings: NSObject {
+class PurchaseStrings {
```

Comment:
I agree this being a class is odd. I gave it a try with IdentityStrings and made the following changes:

Made Identity an `enum` of String

```
enum Identity: String {

        case changing_app_user_id = "Changing App User ID: %@ -> %@",
             logging_in_with_nil_appuserid = """
                                             The appUserID passed to logIn is nil or empty. Can't log in. 
                                             This method should only be called with non-nil and non-empty values.
                                             """,
                                             ```
                                             
To reference it the same way we do at the moment where we do `Strings.identity` I removed it from `Strings.swift` and made it part of a extension of `Strings`.

```
extension Strings {

    enum Identity: String {
```

This changes the way the strings are referenced to:

```
Strings.Identity.changing_app_user_id.rawValue
```

Instead of the current:

```
Strings.identity.changing_app_user_id
```

What do you think? @RevenueCat/sdk

Reply:
@juanpe I'm so glad you brought this up! I had forgotten about it. 

The reason that this was structured this way was that it was the only way I could find that would guarantee Obj-C compatibility and a similar-looking syntax for Swift and Obj-C, without having to make the enums public, and without having to load all the strings to memory at once (that's also why they were computed vars).

I made a mental note-to-self to refactor to enums if we ever migrated the entire codebase to Swift, but at the time I thought that was unlikely to ever happen 😆

And now I got so used to seeing these files that I forgot about it. 

But now that we don't need to reference these from Obj-C, these can be migrated to nice-looking enums 🎉🎉🎉

Reply:
Sounds goods! 👌

You could improve it even more. Since the tokens are constants, you could use constants instead and `rawValue` would be not needed. So, the code’ll look like this:

```swift

enum Numbers {
    static let one = "Uno"
}
print(Numbers.one) // Uno

```

WDYT?

Reply:
I am going to do a followup PR with these changes since it will change a lot of lines 👍